### PR TITLE
feat(oracle): limit max size of price list for each token

### PR DESCRIPTION
**Pull request by leonz789**
## Description

Add limitation for price list size of each token, only keep recent prices instead of all prices for each token.
This is a breaking change which modified proto define of oracle module.

----

Closes #XXX

## Summary by CodeRabbit

- **New Features**
	- Introduced a new parameter `MaxSizePrices` to control the number of price rounds retained.
	- Added functionality to update and validate the `MaxSizePrices` parameter within existing methods.

- **Bug Fixes**
	- Enhanced validation logic to ensure `MaxSizePrices` is at least `1`.

- **Tests**
	- Expanded test coverage to include validations for the new `MaxSizePrices` functionality and parameter updates. 

- **Chores**
	- Improved clarity and organization of existing parameters in relevant files.

### CODE REVIEW

File: x/oracle/keeper/msg_server_update_params_test.go [hunk 1]
```diff
@@ -152,7 +152,20 @@ var _ = Describe("MsgUpdateParams", Ordered, func() {
 
 	})
 
-	Context("", func() {
+	Context("udpate maxSizePrices", func() {
```
Comment (MaxMustermann2) [inline x/oracle/keeper/msg_server_update_params_test.go]:
```suggestion
	Context("update maxSizePrices", func() {
```

File: x/oracle/keeper/params_test.go [hunk 2]
```diff
@@ -215,3 +215,10 @@ func TestUpdateTokenFeeder(t *testing.T) {
 		})
 	}
 }
+
+func TestParamsValidate(t *testing.T) {
+	p := types.DefaultParams()
+	p.MaxSizePrices = 0
+	err := p.Validate()
+	require.ErrorIs(t, err, types.ErrInvalidParams.Wrap("invalid MaxValidators"))
```
Comment (MaxMustermann2) [inline x/oracle/keeper/params_test.go]:
```suggestion
	require.ErrorIs(t, err, types.ErrInvalidParams.Wrap("invalid MaxSizePrices"))
```

**Review by MaxMustermann2 — APPROVED** [review]

File: x/oracle/keeper/params_test.go (hunk 2 above)
Comment (leonz789) [reply x/oracle/keeper/params_test.go]: updated

File: x/oracle/keeper/aggregator/context.go [hunk 3]
```diff
@@ -317,6 +317,10 @@ func (agc *AggregatorContext) GetParams() types.Params {
 	return *agc.params
 }
 
+func (agc *AggregatorContext) GetParamsMaxSizePrices() uint64 {
+	return uint64(agc.params.MaxSizePrices)
+}
```
[the remark this answers is not in the document]
Comment (leonz789) [reply x/oracle/keeper/aggregator/context.go:322]: nil check should be taken care by caller. 'agc' will always be initialized before used

**Review by mikebraver — APPROVED** [review]
LGTM

**Review by bwhour — APPROVED** [review]

**Pull request merged**